webapp/src/components/user_settings/video_devices_settings_section.tsx

```diff
@@ -0,0 +1,446 @@
+        }
+        return () => {
+            navigator.mediaDevices?.removeEventListener('devicechange', loadVideoDevices);
+        };
+    }, [active]);
+
+    if (!active) {
+        return (
+            <div
+                className='section-min'
+                onClick={() => setActive(!active)}
+            >
+                <div className='secion-min__header'>
+                    <h4 className='section-min__title'>
+                        <span>{title}</span>
+                    </h4>
+                    <button
+                        className='color--link style--none section-min__edit'
+                        aria-labelledby=''
+                        aria-expanded={active}
+                    >
+                        <i
+                            className='icon-pencil-outline'
+                            title={editLabel}
+                        />
+                        <span>{editLabel}</span>
+                    </button>
+                </div>
+                <div className='section-min__describe'>
+                    <span>{description}</span>
+                </div>
+            </div>
+        );
+    }
+
+    return (
+        <section className='section-max form-horizontal'>
+            <h4 className='col-sm-12 section-title'>
+                <span>{title}</span>
+            </h4>
+            <div className='sectionContent col-sm-10 col-sm-offset-2'>
+                <div
+                    tabIndex={-1}
+                    className='setting-list'
+                >
+                    <div className='setting-list-item'>
+                        <Fieldset>
+                            <VideoDevicesSelection
+                                deviceType='videoinput'
+                                devices={devices.filter((device) => device.kind === 'videoinput')}
+                                onSelectionChange={async (opt) => {
+                                    previewStream?.getVideoTracks()[0].stop();
+
+                                    const videoOpts = {
+                                        ...DefaultVideoTrackOptions,
+                                    };
+
+                                    if (opt.value !== '') {
+                                        videoOpts.deviceId = {
+                                            exact: opt.value,
+                                        };
+                                    }
+
+                                    const stream = await navigator.mediaDevices.getUserMedia({
+                                        video: videoOpts,
+                                        audio: false,
+                                    });
+
+                                    setPreviewStream(stream);
+                                }}
+                                ref={videoInputsRef}
+                            />
+                            <VideoPreview
```

Comment:
is there a way to reuse components, or is it not worth the trouble?

Reply:
From the widget you mean? They serve very different purposes. Maybe a shared subcomponent but I am not sure.